openstack/utils/choose_version.go

```diff
@@ -59,7 +60,7 @@ func ChooseVersion(client *gophercloud.ProviderClient, recognized []*Version) (*
 	}
 
 	var resp response
-	_, err := client.Request("GET", client.IdentityBase, &gophercloud.RequestOpts{
+	_, err := client.Request(context.Background(), "GET", client.IdentityBase, &gophercloud.RequestOpts{
```

Comment:
Please create a `ChooseVersionWithContext` that accepts a context instead, then add a `ChooseVersion` function for backwards compatibility